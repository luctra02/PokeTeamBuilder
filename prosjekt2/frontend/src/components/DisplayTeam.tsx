import CardComponent from './CardComponent';
import { useNavigate } from 'react-router-dom';
import '../styles/PokemonTeam.css';
import { useQuery } from '@apollo/client';
import { GET_TEAM } from '../graphql/queries';
import { Pokemon } from '../utils/constants';

function DisplayTeam() {
  // Hook for programmatic navigation
  const navigate = useNavigate();

  // Retrieve the team ID from local storage
  const id = localStorage.getItem('teamId');

  // GraphQL query to get team information
  const { loading, data } = useQuery(GET_TEAM, {
    variables: { teamId: id }
  });

  // Loading state check
  if (loading) {
    return <div>Loading...</div>;
  }

  // Function to navigate to the details page of a specific Pokemon
  function changeToDetailPage(pokemon: Pokemon) {
    navigate(`/pokemonInfo/${pokemon.id}`, { state: { pokemon } });
  }

  // JSX structure for displaying the team
  return (
    <div className="teamContainer"> 
      <section className="pokemonTeamDisplayBox">
        {/* Map through the team's Pokemon and display each in a card */}
        {data?.getTeam.pokemon.map((pokemon: Pokemon) => (
          <article className="pokemonDisplayButton" key={pokemon.id} onClick={() => changeToDetailPage(pokemon)}>
            <CardComponent
              pokemonObject={{
                id: pokemon.id,
                name: pokemon.name,
                image: pokemon.image,
                types: pokemon.types,
                baseStats: pokemon.baseStats,
                weight: pokemon.weight,
                height: pokemon.height,
              }}
            />
          </article>
        ))}
      </section>
    </div>
  );
}

export default DisplayTeam;
