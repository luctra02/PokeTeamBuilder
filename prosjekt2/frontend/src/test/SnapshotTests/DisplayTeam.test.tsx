import { render, screen, waitFor } from '@testing-library/react';
import { allMocks as mocks} from '../mocks/mockQueries';
import { testPageRender } from '../testUtils';
import DisplayTeam from '../../pages/DisplayTeamPage';




describe('renders a team', () => {
    it('matches the snapshot', async () => {
      const { asFragment } = testPageRender(<DisplayTeam/>, { mocks });
        
       await waitFor(() => {
        localStorage.setItem('teamId', "8fpgpxy2x")
        expect(screen.getByText(/Pikachu/i)).toBeInTheDocument();
        expect(screen.getByText(/Remove from Team/i)).toBeInTheDocument();
      });
  
      expect(asFragment()).toMatchSnapshot();
    });
  }); 
