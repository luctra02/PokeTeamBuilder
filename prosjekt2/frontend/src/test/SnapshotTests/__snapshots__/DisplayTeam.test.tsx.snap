// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders a team > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="teamContainer"
  >
    <div
      class="pokemonTeamDisplayBox"
    >
      <div
        class="pokemonDisplayButton"
      >
        <div
          class="pokemonCard"
        >
          <h2>
            Pikachu
          </h2>
          <img
            alt=""
            src="https://play.pokemonshowdown.com/sprites/ani/pikachu.gif"
          />
          <div
            class="pokemonID"
          >
            #0025
          </div>
          <div
            class="pokeType"
          >
            <div>
              <div
                class="pokeType PokeTypeBox"
                style="background-color: rgb(248, 208, 48);"
              >
                Electric
              </div>
            </div>
          </div>
          <button
            class="teamButton"
          >
            Remove from Team
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
